test/functional/wallet_inactive_hdchains.py

```diff
@@ -0,0 +1,193 @@
+        assert test_wallet.getaddressinfo(addr2)["ismine"]
+
+        # Send to second address on the old seed
+        txid = default.sendtoaddress(addr2, 10)
+        self.generate(self.nodes[0], 1)
+        test_wallet.gettransaction(txid)
+
+    def test_encrypted_wallet(self):
+        self.log.info("Test inactive HD chains when wallet is encrypted")
+        default = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+
+        self.nodes[0].createwallet(wallet_name="enc_base", descriptors=False, blank=True)
+        self.nodes[0].createwallet(wallet_name="enc_test", descriptors=False, blank=True)
+        base_wallet = self.nodes[0].get_wallet_rpc("enc_base")
+        test_wallet = self.nodes[0].get_wallet_rpc("enc_test")
+
+        # Setup both wallets with the same HD seed
+        seed = get_generate_key()
+        base_wallet.sethdseed(True, seed.privkey)
+        test_wallet.sethdseed(True, seed.privkey)
+
+        # Encrypting will generate a new HD seed and flush the keypool
+        test_wallet.encryptwallet("pass")
+
+        # Get the address at index 6
+        for _ in range(0, 5):
+            base_wallet.getnewaddress()
+        addr1 = base_wallet.getnewaddress()
+        assert test_wallet.getaddressinfo(addr1)["ismine"]
+        # Get the address at index 13
+        for _ in range(0, 6):
+            base_wallet.getnewaddress()
+        addr2 = base_wallet.getnewaddress()
+        assert not test_wallet.getaddressinfo(addr2)["ismine"]
+
+        # Send to first address on the old seed
+        txid = default.sendtoaddress(addr1, 10)
+        self.generate(self.nodes[0], 1)
+        test_wallet.gettransaction(txid)
+
+        # The test wallet will not be able to generate the topped up keypool
+        # until it is unlocked. So it still should not know about the second address
+        assert not test_wallet.getaddressinfo(addr2)["ismine"]
+        test_wallet.walletpassphrase("pass", 1)
+
+        # The test wallet should now know about the second address as it
+        # should have generated it in the inactive chain's keypool
+        assert test_wallet.getaddressinfo(addr2)["ismine"]
+
+        # Send to second address on the old seed
+        txid = default.sendtoaddress(addr2, 10)
+        self.generate(self.nodes[0], 1)
+        test_wallet.gettransaction(txid)
+
+    def test_without_upgraded_keymeta(self):
+        # Test that it is possible to top up inactive hd chains even if there is no key origin
+        # in CKeyMetadata. This tests for the segfault reported in
+        # https://github.com/bitcoin/bitcoin/issues/21605
+        self.log.info("Test that topping up inactive HD chains does not need upgraded key origin")
+        default = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+
+        self.nodes[0].createwallet(wallet_name="keymeta_base", descriptors=False, blank=True)
+        self.nodes[1].createwallet_passthrough(wallet_name="keymeta_test")
```

Comment:
I assume this is important, probably needs a comment on what this is doing because I don't understand it

Reply:
Added a comment.

It's because we override createwallet in the test framework so that passing `--descriptors` will autofill the descriptor parameter. But that doesn't work for old nodes, so we need a passthrough function.